examples/GroupCharacters.jl

```diff
@@ -0,0 +1,348 @@
+    return QabModule.QabElem(val,n)
+end
+
+#function QabModule.QabElem(cyc::GAP.GapObj, c::Int)
+#end
+
+function gap_cyclotomic(elm::QabModule.QabElem)
+    coeffs = [Nemo.coeff(elm.data, i) for i in 0:(elm.c-1)]  # fmpq
+    coeffs = [GAP.julia_to_gap(BigInt(numerator(x)) // BigInt(denominator(x)))
+              for x in coeffs]
+    return GAP.Globals.CycList(GAP.julia_to_gap(coeffs))
+end
+
+
+#############################################################################
+##
+##  character tables
+##
+abstract type GroupCharacterTable end
+
+mutable struct GAPGroupCharacterTable <: GroupCharacterTable
+    GAPGroup::Oscar.GAPGroup    # the underlying group, if any
+    GAPTable::GAP.GapObj  # the character table object
+    characteristic::Int
+    AbstractAlgebra.@declare_other
+
+    function GAPGroupCharacterTable(G::Oscar.GAPGroup, tab::GAP.GapObj, char::Int)
+      ct = new()
+      ct.GAPGroup = G
+      ct.GAPTable = tab
+      ct.characteristic = char
+      return ct
+    end
+
+    function GAPGroupCharacterTable(tab::GAP.GapObj, char::Int)
+      ct = new()
+      #ct.GAPGroup is left undefined
+      ct.GAPTable = tab
+      ct.characteristic = char
+      return ct
+    end
+end
+
+function character_table(G::Oscar.GAPGroup, p::Int = 0)
+    tbl = AbstractAlgebra.get_special(G, :character_table)
+    if tbl == nothing
+      tbl = GAP.Globals.CharacterTable(G.X)
+      AbstractAlgebra.set_special(G, :character_table => tbl)
+    end
+
+    if p != 0
+      isprime(p) || error("p must be 0 or a prime integer")
+      tbl = mod(tbl, p)
+    end
+
+    return GAPGroupCharacterTable(G, tbl, p)
+end
+
+function character_table(id::String, p::Int = 0)
+    if GAP.Globals.IsBoundGlobal(GAP.julia_to_gap("CTblLib"))
+      tbl = GAP.Globals.CharacterTable(GAP.julia_to_gap(id))
+      if tbl == GAP.Globals.fail
+        return nothing
+      elseif p != 0
+        isprime(p) || error("p must be 0 or a prime integer")
+        tbl = GAP.Globals.mod(tbl, GAP.julia_to_gap(p))
+        if tbl == GAP.Globals.fail
+          return nothing
+        end
+      end
+      return GAPGroupCharacterTable(tbl, p)
+    else
+      error("no character table library available")
+    end
+end
+
+function Base.show(io::IO, tbl::GAPGroupCharacterTable)
+    print(io, GAP.CSTR_STRING(GAP.Globals.StringDisplayObj(tbl.GAPTable)))
+end
+
+function Base.print(io::IO, tbl::GAPGroupCharacterTable)
+    gaptbl = tbl.GAPTable
+    if GAP.Globals.HasUnderlyingGroup(gaptbl)
+      id = string(GAP.Globals.UnderlyingGroup(gaptbl))
+    else
+      id = "\"" * String(GAP.Globals.Identifier(gaptbl)) * "\""
+    end
+    print(io, "character_table($id)")
+end
+
+Base.length(tbl::GAPGroupCharacterTable) = return length(GAP.Globals.Irr(tbl.GAPTable))
+AbstractAlgebra.nrows(tbl::GAPGroupCharacterTable) = return length(GAP.Globals.Irr(tbl.GAPTable))
+AbstractAlgebra.ncols(tbl::GAPGroupCharacterTable) = return length(GAP.Globals.Irr(tbl.GAPTable))
```

Comment:
```suggestion
Base.length(tbl::GAPGroupCharacterTable) = length(GAP.Globals.Irr(tbl.GAPTable))
AbstractAlgebra.nrows(tbl::GAPGroupCharacterTable) = length(GAP.Globals.Irr(tbl.GAPTable))
AbstractAlgebra.ncols(tbl::GAPGroupCharacterTable) = length(GAP.Globals.Irr(tbl.GAPTable))
```

Naive question: Is the "length" of a character table a standard notion? Personally (as a non-expert) I'd rather think of the "size" or perhaps the number of rows and columns (which nicely matches with those nrows/ncrols methods).

Ahh, perhaps this is required for getindex to work?

Reply:
Now I think that introducing `length` for character tables would be confusing in Julia: For a matrix, `length` returns the number of entries.
I will remove this method.